lms/services/assignment.py

```diff
@@ -59,6 +59,26 @@ def set_document_url(
         # just one key from the cache, you have to clear the entire cache.)
         self.get.cache_clear()
 
+    def get_canvas_mapped_file_id(
+        self, tool_consumer_instance_guid, resource_link_id, file_id
+    ):
+        mic = self.get(tool_consumer_instance_guid, resource_link_id)
+
+        if not mic:
+            return None
+
+        return mic.extra.get("canvas_file_mappings", {}).get(file_id)
+
+    def set_canvas_mapped_file_id(
+        self, tool_consumer_instance_guid, resource_link_id, file_id, mapped_file_id
+    ):
+        mic = self.get(tool_consumer_instance_guid, resource_link_id)
+
+        if not mic:
+            raise ValueError("ModuleItemConfiguration not found")
+
+        mic.extra.setdefault("canvas_file_mappings", {})[file_id] = mapped_file_id
```

Comment:
I think it's probably simplest to just mix the Canvas-specific methods in with the non-specific methods in `AssignmentService` for now. But services make it easy to separate them in future if we want. You can have separate services and they can use each other via composition. So for example you could have a separate `CanvasFilesAssignmentService`. The service factory can even return different services depending on the context